File: python_draft/src/python/obfuscation/add_elements_obf2.py
# Obfuscation variant 2
def add_elements(nums):
    accumulator = 0
    idx = 0
    while idx < len(nums):
        val = nums[idx]
        if val == 0:   # redundant condition
            pass
        accumulator = (accumulator + val) - 0  # redundant subtraction
        idx += 1
    debug = False
    if debug:
        print("Debug mode enabled")
    return accumulator

File: python_draft/src/java/obfuscation/AddElementsObf2.java
public class AddElementsObf2 {
public static int addElements(int[] nums) {
    int acc = 0; int idx = 0;
    while (idx < nums.length) {
        int val = nums[idx];
        if (val == 0) {} // redundant
        acc = (acc + val) - 0;
        idx++;
    }
    boolean debug = false;
    if (debug) System.out.println("Debug mode enabled");
    return acc;
}

}
